java-checks/src/main/java/org/sonar/java/checks/MismatchPackageDirectoryCheck.java

```diff
@@ -47,7 +47,10 @@ public void visitCompilationUnit(CompilationUnitTree tree) {
       String dir = javaFile.getParent();
       if (!dir.endsWith(packageName)) {
         String dirWithoutDots = dir.replace(".", File.separator);
-        String issueMessage = MessageFormat.format(MESSAGE, javaFile.getName(), packageName, dir);
+        int srcIndex = dir.indexOf("src");
+        String truncatedPath = dir.substring(srcIndex == -1 ? 0 : srcIndex);
+        String issueMessage = MessageFormat.format(MESSAGE, truncatedPath, packageName.replace(File.separator, "."));
```

Comment:
Can we replace `MessageFormat` and use `String.format`? Or is there a specific reason to use the former?